core-bundle/src/Filesystem/MountManager.php

```diff
@@ -402,12 +402,14 @@ private function getAdapterAndPath(string $path): array
     {
         $prefix = $path;
 
-        // Find adapter with the longest (= most specific) matching prefix
-        do {
-            if (null !== ($adapter = $this->mounts[$prefix] ?? null)) {
-                return [$adapter, Path::makeRelative($path, $prefix), $prefix];
-            }
-        } while ('.' !== ($prefix = \dirname($prefix)));
+        if ('' !== $path) {
```

Comment:
Should `$prefix = $path;` be moved inside the `if` block?

Reply:
See 1ffee24e16e05bbf9bbfa30bb01c2ba9c01e4338.